### PR TITLE
[et] Prepare local_engine.json for CI, teach et to understand local build names

### PR DESCRIPTION
**Pull request by zanderso**
This is steps (2) and (3) of https://github.com/flutter/flutter/issues/145263.

The next steps after this are to:
1. Fix any issues that come up when running `local_engine.json` in CI.
1. Step (4) of https://github.com/flutter/flutter/issues/145263
1. Fill in some missing builds in `local_engine.json`.

### CODE REVIEW

File: tools/engine_tool/lib/src/build_utils.dart [hunk 1]
```diff
@@ -65,6 +65,31 @@ void debugCheckBuilds(List<Build> builds) {
   }
 }
 
+/// Transform the name of a build into the name presented and accepted by the
+/// CLI
+///
+/// If a name starts with '$OS/', it is a local development build, and the
+/// mangled name has the '$OS/' part stripped off.
+///
+/// If the name does not start with '$OS/', then it starts with 'ci/' and is
```
Comment (matanlurey) [inline tools/engine_tool/lib/src/build_utils.dart:74]: Do we want to throw if it doesn't?

File: tools/engine_tool/lib/src/build_utils.dart [hunk 2]
```diff
@@ -65,6 +65,31 @@ void debugCheckBuilds(List<Build> builds) {
   }
 }
 
+/// Transform the name of a build into the name presented and accepted by the
+/// CLI
+///
+/// If a name starts with '$OS/', it is a local development build, and the
```
Comment (matanlurey) [inline tools/engine_tool/lib/src/build_utils.dart:71]: Can we expand this with 1 more sentence of examples of `$OS` (maybe mention it's `[Platorm.operatingSystem]`).

File: tools/engine_tool/lib/src/build_utils.dart [hunk 3]
```diff
@@ -65,6 +65,31 @@ void debugCheckBuilds(List<Build> builds) {
...
+}
+
+/// Transform the mangled (CLI) name of a build into its true name in the build
+/// config json file.
+///
+/// This does the reverse of [mangleConfigName] taking the operating system
+/// name from `environment`.
+String demangleConfigName(Environment environment, String name) {
+  const String cif = 'ci/';
+  const String cib = r'ci\';
```
Comment (matanlurey) [inline tools/engine_tool/lib/src/build_utils.dart:88]: Is this for Windows or something else, I don't understand this one :P

File: tools/engine_tool/lib/src/build_utils.dart [hunk 4]
```diff
@@ -65,6 +65,41 @@ void debugCheckBuilds(List<Build> builds) {
...
 
+/// Transform the name of a build into the name presented and accepted by the
+/// CLI
+///
+/// If a name starts with '$OS/', it is a local development build, and the
+/// mangled name has the '$OS/' part stripped off, where $OS is the value of
+/// `platform.operatingSystem` in the passed-in `environment`.
+///
+/// If the name does not start with '$OS/', then it must start with 'ci/' or
+/// 'ci\' in which case the name is returned unchanged.
```
Comment (johnmccutchan) [inline tools/engine_tool/lib/src/build_utils.dart:76]: some example inputs/outputs would help this comment

File: tools/engine_tool/lib/src/build_utils.dart (hunk 3 above)
Comment (johnmccutchan) [reply tools/engine_tool/lib/src/build_utils.dart:98]: Yes, I would rename cif and cib to something that indicates Win/Unix path endings.

Also you could factor these constants out and use them above on line 82.

**Review by johnmccutchan — APPROVED** [review]

File: tools/engine_tool/lib/src/build_utils.dart (hunk 1 above)
Comment (johnmccutchan) [reply tools/engine_tool/lib/src/build_utils.dart]: +1

File: tools/engine_tool/lib/src/build_utils.dart (hunk 3 above)
Comment (matanlurey) [reply tools/engine_tool/lib/src/build_utils.dart:98]: If so I think we could use `package:path/path.dart#fragment` (or just rely on the methods in `path` that implicitly use the right fragment).

File: tools/engine_tool/lib/src/build_utils.dart (hunk 2 above)
Comment (zanderso) [reply tools/engine_tool/lib/src/build_utils.dart:79]: Done.

File: tools/engine_tool/lib/src/build_utils.dart (hunk 1 above)
Comment (zanderso) [reply tools/engine_tool/lib/src/build_utils.dart]: Done.

File: tools/engine_tool/lib/src/build_utils.dart (hunk 3 above)
Comment (zanderso) [reply tools/engine_tool/lib/src/build_utils.dart]: Cleaned this up using these suggestions. Thanks!

File: tools/engine_tool/lib/src/build_utils.dart (hunk 4 above)
Comment (zanderso) [reply tools/engine_tool/lib/src/build_utils.dart]: Done.

**Pull request merged**